src/transaction/boot_cl.c

```diff
@@ -911,7 +911,12 @@ boot_restart_client (BOOT_CLIENT_CREDENTIAL * client_credential)
     {
       if (au_has_user_name ())
 	{
-	  client_credential->db_user = au_user_name ();
+	  char *name = db_get_user_name ();
+	  if (name != NULL)
+	    {
+	      client_credential->db_user = name;
+	      ws_free_string (name);
+	    }
```

Comment:
please confirm the block: it points and frees.

Reply:
db_user is std::string and will make and keep a copy of name, therefore a temporary was leaked before.